lib/common/socket/evloop.c.h

```diff
@@ -399,19 +399,21 @@ static struct st_h2o_evloop_socket_t *create_socket(h2o_evloop_t *loop, int fd,
     return sock;
 }
 
-static struct st_h2o_evloop_socket_t *create_socket_set_nodelay(h2o_evloop_t *loop, int fd, int flags)
+static struct st_h2o_evloop_socket_t *create_socket_set_nodelay(h2o_evloop_t *loop, int fd, int flags, int sa_family)
 {
     /* ignore errors returned by setsockopt; fd is not restricted to TCP sockets (could be a unix-domain socket for example) */
-    int on = 1;
-    setsockopt(fd, IPPROTO_TCP, TCP_NODELAY, &on, sizeof(on));
+    if (sa_family != AF_UNIX) {
+        int on = 1;
+        setsockopt(fd, IPPROTO_TCP, TCP_NODELAY, &on, sizeof(on));
+    }
```

Comment:
I believe that this `if` is an approximation of `if (proto == IPPROTO_TCP)`?

Reducing the number of unnecessary function calls is fine, but I am concerned about the long-term impact of designing the API based on approximations.

How about removing `create_socket_set_nodelay` as a whole, and doing the following? All the comments below are based on this assumption.